helix-core/src/main/java/org/apache/helix/controller/stages/BestPossibleStateCalcStage.java

```diff
@@ -282,22 +263,22 @@ private boolean validateOfflineInstancesLimit(final ResourceControllerDataProvid
 
     Map<String, IdealState> newIdealStates = new HashMap<>();
 
-    ClusterConfig clusterConfig = cache.getClusterConfig();
-    WagedRebalancer wagedRebalancer =
-        getWagedRebalancer(helixManager, clusterConfig.getGlobalRebalancePreference(),
-            clusterConfig.isGlobalRebalanceAsyncModeEnabled());
-    try {
-      newIdealStates.putAll(wagedRebalancer.computeNewIdealStates(cache, wagedRebalancedResourceMap,
-          currentStateOutput));
-    } catch (HelixRebalanceException ex) {
-      // Note that unlike the legacy rebalancer, the WAGED rebalance won't return partial result.
-      // Since it calculates for all the eligible resources globally, a partial result is invalid.
-      // TODO propagate the rebalancer failure information to updateRebalanceStatus for monitoring.
+    if (wagedRebalancer != null) {
```

Comment:
Perhaps we should check whether the rebalancer has been closed here?

Reply:
As mentioned above, I would prefer to reject call (by throwing exceptions or so) inside the rebalancer. For the calc stage, there is nothing it can do.
Moreover, checking inside the rebalancer can help to keep the close status private.
TODO added.

Reply:
Where is the reject call? Are you referring to Line 271?

Reply:
No reject call yet. TODO is for this task. It is not a must for now since all the callers are private.

Reply:
I still don't see a TODO stating that we need to check on the rebalancer status (whether it's closed, null, etc.)